nginx/ngx_js_shared_dict.c

```diff
@@ -765,7 +767,32 @@ njs_js_ext_shared_dict_incr(njs_vm_t *vm, njs_value_t *args, njs_uint_t nargs,
         njs_value_number_set(init, 0);
     }
 
-    rc = ngx_js_dict_incr(shm_zone->data, &key, delta, init, &value);
+    timeout_src = njs_arg(args, nargs, 4);
+    if (!njs_value_is_undefined(timeout_src)) {
+
+        if (!njs_value_is_number(timeout_src)) {
+            njs_vm_type_error(vm, "timeout is not a number");
+            return NJS_ERROR;
+        }
+
+        timeout = (ngx_msec_t) njs_value_number(timeout_src);
+
+        if (timeout < 1) {
+            njs_vm_type_error(vm,
+                              "timeout must be greater than or equal to 1");
+            return NJS_ERROR;
+        }
+
+        if (!dict->timeout) {
```

Comment:
Probably we should check that before timeout extraction.

Reply:
Agreed, makes sense.